### PR TITLE
feat: introduce hops-debug as a node/browser wrapper for debug

### PR DESCRIPTION
**Pull request by ZauberNerd**
This package wraps the `debug` module to allow smaller client-side
bundles, because we do not include `debug` in the client build.

This reduces our bundles by about 6kb.

<details>
<summary>Bors merge bot cheat sheet</summary>

We are using [bors-ng](https://github.com/bors-ng/bors-ng) to automate merging of our pull requests. The following table provides a summary of commands that are available to reviewers (members of this repository with push access) and delegates (in case of `bors delegate+` or `bors delegate=[list]`).

| Syntax | Description |
| --- | --- |
| bors merge | Run the test suite and push to master if it passes. Short for "reviewed: looks good." |
| bors merge- | Cancel an r+, r=, merge, or merge= |
| bors try | Run the test suite without pushing to master. |
| bors try- | Cancel a try |
| bors delegate+ | Allow the pull request author to merge their changes. |
| bors delegate=[list] | Allow the listed users to r+ this pull request's changes. |
| bors retry | Run the previous command a second time. |

This is a short collection of opinionated commands. For a full list of the commands read the [bors reference](https://bors.tech/documentation/).

</details>

### CODE REVIEW

File: packages/debug/browser.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// eslint-disable-next-line no-unused-vars
+module.exports = function debug(_name) {
+  // eslint-disable-next-line no-unused-vars
+  return function (..._args) {};
```
Comment (KnisterPeter) [inline packages/debug/browser.js:4]: Don't you want to use `console` here?

Comment (herschel666) [reply packages/debug/browser.js:4]: Yeah, what stops us from doing `console.debug(name, ...args)` here?

Comment (ZauberNerd) [reply packages/debug/browser.js:4]: We could do that. I just thought that we've never really used the `debug` module in the browser and therefore we could just replace it with a no-op.

Comment (KnisterPeter) [reply packages/debug/browser.js:4]: Both would be fine for me

Comment (herschel666) [reply packages/debug/browser.js:4]: It might provide value for the users. But I have no strong opinion here.

**Comment by herschel666** [comment]
Did you intentionally not create a README for the new package?

File: packages/debug/browser.js (hunk 1 above)
Comment (ZauberNerd) [reply packages/debug/browser.js]: I've implemented a simple solution based on cookies and `console.debug`.

**Comment by ZauberNerd** [comment]
> Did you intentionally not create a README for the new package?

I've now added a readme. Please take another look.

File: packages/debug/browser.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/* eslint-env browser */
+
+const debugCookie =
+  document.cookie.match('(^|;)\\s*hops_debug\\s*=\\s*([^;]+)')?.pop() || '';
```
Comment (herschel666) [inline packages/debug/browser.js:4]: Is there a reason why the regex is a string and not a RegExp shorthand?

File: packages/debug/browser.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+const debugRegex = new RegExp(debugCookie.replace('*', '.*'));
+
+module.exports = function debug(name) {
+  const isEnabled = debugCookie !== '' && debugRegex.test(name);
+  console.log({ name, debugCookie, debugRegex, isEnabled });
```
Comment (herschel666) [inline packages/debug/browser.js:9]: You forgot to remove this. :)

File: packages/debug/browser.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            `%c${name}%c`,
+            'font-weight: bold',
+            'font-weight: initial',
+            format,
+            ...args
+          );
+        }
+      }
+    : // eslint-disable-next-line no-unused-vars
+      (..._args) => {};
```
Comment (herschel666) [inline packages/debug/browser.js:30]:
```suggestion
      () => {};
```

**Review by herschel666 — CHANGES_REQUESTED** [review]

File: packages/debug/browser.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+/* eslint-env browser */
+
+const debugCookie =
+  (document.cookie.match(/(^|;)\\s*hops_debug\\s*=\\s*([^;]+)/) || []).pop() ||
```
Comment (herschel666) [inline packages/debug/browser.js:4]:
```suggestion
  (document.cookie.match(/(^|;)\s*hops_debug\s*=\s*([^;]+)/) || []).pop() ||
```

No need escape backslashes anymore.

**Review by herschel666 — APPROVED** [review]

File: packages/debug/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+If you don't already have an existing Hops project read this section [on how to set up your first Hops project.](../../DOCUMENTATION.md#quick-start)
+
+### Usage
+
+`import` / `require` this package to create a debugger:
+
+```javascript
+import debugFactor from 'hops-debug';
```
Comment (herschel666) [inline packages/debug/README.md:26]:
```suggestion
import debugFactory from 'hops-debug';
```

**Review by herschel666 — DISMISSED** [review]

**Review by herschel666 — APPROVED** [review]

**Comment by ZauberNerd** [comment]
bors merge

**Pull request merged**